src/extensions/step_handlers/swupdate_handler/src/swupdate_handler.cpp

```diff
@@ -278,38 +290,50 @@ ADUC_Result SWUpdateHandlerImpl::Install(const tagADUC_WorkflowData* workflowDat
 ADUC_Result SWUpdateHandlerImpl::Apply(const tagADUC_WorkflowData* workflowData)
 {
     ADUC_Result result = { ADUC_Result_Failure };
-    char* workFolder = NULL;
+    const ADUC_ConfigInfo* config = nullptr;
+    char* workFolder = nullptr;
 
     if (workflow_is_cancel_requested(workflowData->WorkflowHandle))
     {
         return this->Cancel(workflowData);
     }
 
+    config = ADUC_ConfigInfo_GetInstance();
+    if (config == nullptr)
+    {
+        Log_Error("Failed to get config info instance.");
+        result.ExtendedResultCode = ADUC_ERC_SWUPDATE_HANDLER_APPLY_FAILED_TO_GET_CONFIG_INSTANCE;
+        goto done;
+    }
+
     workFolder = workflow_get_workfolder(workflowData->WorkflowHandle);
     Log_Info("Applying data from %s", workFolder);
 
-    // Execute the install command with  "-a" to apply the install by telling
-    // the bootloader to boot to the updated partition.
+    {
+        // Execute the install command with  "-a" to apply the install by telling
+        // the bootloader to boot to the updated partition.
 
-    // This is equivalent to : command << c_installScript << " -l " << _logFolder << " -a"
+        // This is equivalent to : command << c_installScript << " -l " << _logFolder << " -a"
 
-    std::vector<std::string> args{ adushconst::update_type_opt,
-                                   adushconst::update_type_microsoft_swupdate,
-                                   adushconst::update_action_opt,
-                                   adushconst::update_action_apply };
+        std::vector<std::string> aduShellArgs{
+            adushconst::config_folder_opt, config->configFolder,
+            adushconst::update_type_opt,   adushconst::update_type_microsoft_swupdate,
+            adushconst::update_action_opt, adushconst::update_action_apply
+        };
 
-    args.emplace_back(adushconst::target_log_folder_opt);
-    args.emplace_back(ADUC_LOG_FOLDER);
+        aduShellArgs.emplace_back(adushconst::target_log_folder_opt);
+        aduShellArgs.emplace_back(ADUC_LOG_FOLDER);
 
-    std::string output;
+        std::string output;
 
-    const int exitCode = ADUC_LaunchChildProcess(ADUSHELL_FILE_PATH, args, output);
+        const int exitCode = ADUC_LaunchChildProcess(config->aduShellFilePath, aduShellArgs, output);
 
-    if (exitCode != 0)
-    {
-        Log_Error("Apply failed, extendedResultCode = %d", exitCode);
-        result = { ADUC_Result_Failure, exitCode };
-        goto done;
+        if (exitCode != 0)
+        {
+            Log_Error("Apply failed, extendedResultCode = %d", exitCode);
+            result = { ADUC_Result_Failure, exitCode };
```

Comment:
Should use MAKE_ADUC_EXTENDEDRESULTCODE_FOR_FACILITY_ADUC_FACILITY_UNKNOWN(ERRNO, exitCode)

Reply:
Fix. Returning ADUC_ERC_SWUPDATE_HANDLER_CHILD_FAILURE_PROCESS_EXITCODE